0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -120,13 +139,14 @@ The response body is a JSON object that includes basic account details necessary
-| `minimum_destination_amount` | Integer String | Minimum amount, denoted in the minimum divisible units of the ledger, the receiver will accept. This amount may be determined by the minimum transfer amount of the ledger. If the receiver expects a specific amounts to be delivered, this may be set equal to the `maximum_destination_amount` |
-| `ledger_info` | Object | Details about the destination ledger |
-| `ledger_info.currency_code` | String | Currency code to identify the receiver's currency. Currencies that have [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) codes should use those. Sender UIs SHOULD be able to render non-standard codes |
-| `ledger_info.currency_scale` | Integer | The scale of the amounts on the destination ledger (e.g. an amount of `"1000"` with a scale of `2` translates to `10.00` units of the destination ledger's currency) |
-| `receiver_info` | Object | Arbitrary additional information about the receiver. This field has no schema and the receiver may include any fields they choose. The field names listed below are recommended merely for interoperability purposes. |
+| `shared_secret` | 32 bytes, [base64 encoded](https://en.wikipedia.org/wiki/Base64) (including padding) | The shared secret to be used by this specific http client in the [Pre-Shared Key protocol](../0016-pre-shared-key/0016-pre-shared-key.md). Should be shared only by the server and this specific http client, and should therefore be different in each query response. |
+| `balance`  | Object | _(OPTIONAL)_ Details of this receiver's balance. Used for invoices and similar temporary accounts. |
+| `balance.maximum` | Integer String | Maximum amount, denoted in the minimum divisible units of the ledger, the receiver will accept. This represents the highest sum that incoming chunks are allowed to reach, not the highest size of an individual chunk (which is determined by path MTU). If this is an invoice the `balance.maximum` is the amount at which the invoice would be considered paid. |
+| `balance.current` | Integer String | Current sum of all incoming chunks. |
+| `ledger_info` | Object | _(OPTIONAL)_ Details about the destination ledger, for sender's display purposes. |
+| `ledger_info.asset_code` | String | Asset code to identify the receiver's currency. Currencies that have [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) codes should use those. Sender UIs SHOULD be able to render non-standard codes |
+| `ledger_info.asset_scale` | Integer | The scale of the amounts on the destination ledger (e.g. an amount of `"1000"` with a scale of `2` translates to `10.00` units of the destination ledger's asset/currency) |
```

Comment:
Why the rename from currency_{code,scale} to asset_{code,scale}? Did the meaning stay the same?

I think these fields should now be optional, if defined at all. In PSK2, there no longer is a concept of destination ledger unit pegging (currency_code/currency_scale), because there are no more curves in route broadcasts, right? Destination ledger units are now only relevant for expressing the percentage of an invoice that is paid by one chunk. Also, they were already a problem because they allow the receiver to manipulate them, and pretend to be a cheap shop on an expensive network.

Reply:
Those fields are now optional, and I've renamed from `ledger_info` to `asset_info`. I see your point about manipulation by the receiver, but I'm not seeing it as an attack vector unless the sender doesn't check the source amount at all. It could make the network as a whole look bad, but that will always be a problem when you quote by destination amounts because the receiver could always put their receiver after a highly expensive connector which they control.